Library/Homebrew/brew.sh

```diff
@@ -6,16 +6,13 @@ case "$HOMEBREW_SYSTEM" in
 esac
 
 # Force UTF-8 to avoid encoding issues for users with broken locale settings.
-if ! command -v locale &>/dev/null
-then
-  export LC_ALL=C
-elif [[ "$(locale charmap 2>/dev/null)" != "UTF-8" ]]
+if [[ "$(locale charmap 2>/dev/null)" != "UTF-8" ]]
```

Comment:
I'm not a fan of suppressing all error messages from `locale` in this way.

Reply:
Do you suggest removing 2>/dev/null here?

Reply:
Can you explain why @sjackman? When we're just using the output to decide on a variable and we have an `else`: what's the upside of showing errors here?

Reply:
For troubleshooting. We do not expect `locale` to fail, and if it were to fail for some unexpected reason, and the user were to report a bug, we had have no basis on which to troubleshoot that big, because the stderr was unconditionally suppressed, which is the shell equivalent of catching all exceptions and ignoring them.

Reply:
We're not checking the exit code of `locale` as-is just checking (twice) if the output contains a string. It doesn't matter to us whether the output doesn't contain that string due to a failure or due to the locale being missing. Considering how we've been printing these warnings at users in the past and causing unnecessary confusion: I'd rather we didn't continue to do so.

Reply:
We have a difference of opinion here, and that's okay. I prefer not to suppress stderr, but I'm fine merging either.